spark/src/main/java/org/apache/iceberg/actions/CreateAction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Map;
+
+interface CreateAction extends Action<Long> {
+
+  /**
+   * Adds additional properties to the newly created Iceberg Table. Any properties with
+   * the same key name will be overwritten.
+   * @param properties a map of properties to be included
+   * @return this for chaining
+   */
+  CreateAction withAdditionalProperties(Map<String, String> properties);
+
+  /**
+   * Adds an additional property to the newly created Iceberg Table. Any properties
+   * with the same key name will be overwritten.
+   * @param key the key of the property to add
+   * @param value the value of the property to add
+   * @return this for chaining
+   */
+  CreateAction withAdditionalProperty(String key, String value);
```

Comment:
Why not `set` and `setAll`? Is "additional" more clear?

Reply:
This was actually part of an earlier discussion I had with @aokolnychyi, We were discussing whether we should be preserving the properties set in the origin table for the operation. I had "additional" here because I settled on a behavior we had previously be using internally which copied over whatever properties the original source originally had. 

I can go back to "set and setAll" but I would like to hear your opinion on copying properties from the origins.

Reply:
I think both migrate and snapshot should copy everything from the original table, including properties. That's one reason why I like `set`: it makes no guarantees about the other table's properties, only that the given key/value will be set in the new table.